File: codingsample/20200813/Smooth_sailing_2_commonCharacterCount.cpp
#include <stdio.h>
//두 문자열이 주어졌을때, 두 문자열에서 공통 문자수를 반환하는 함수 
int commonCharacterCount(char * s1, char * s2) {
	int s1_size , s2_size;
	int empty[100] = {0,};
	int output=0;
	
	for(int i=0;s1[i]!='\0';i++){
		for(int j=0;s2[j]!='\0';j++){
			if(s1[i]==s2[j] && empty[j]==0) {
				output++;
				empty[j]=1;
				break;
			}
		}
	}
	return output;
}
//나의 답안 

int commonCharacterCount2(char * s1, char * s2) {
    int count = 0;
    for(int i = 0; i < strlen(s1); i++){
        for(int j = 0; j < strlen(s2); j++){
            if(s1[i] == s2[j]){
                s1[i] = '#';
                s2[j] = '#';
                count++;
                break;
            }
        }
    }
    return count;
}
//모범답안
//포인트
//문자열 자체에서 걸러버리면 empty같은거 생성안해줘도되는구나 

int main() {
	char a[10] = "aabcc";
	char b[6] = "adcaa";
	int c = commonCharacterCount(a,b);
	printf("%d\n",c);
} 

File: ~2020.11.26 - codingsample/202008/20200813/Smooth_sailing_2_commonCharacterCount.py
def commonCharacterCount(s1, s2):
    return sum(min(s1.count(x), s2.count(x)) for x in set(s1))

def commonCharacterCount2(s1, s2):
    count = 0
    commons = set(s1) & set(s2)
    for i in commons:
        count += min(s1.count(i), s2.count(i))
    return count

# 집합을 이용한 중복값의 제거
# 일반적 답안인 2번은 count변수를 이용하고,
# commons에서 교집합을 이용한다.
# 교집합에 count 함수를 이용하여, s1에서 해당 교집합의 원소값, s2에서 해당 교집합의 원소값을 세며
# min을 통해 두 원소값에서 최솟값을 뽑아냄(같은 원소의 경우 최소값이 총 개수에 해당하기 때문)
# 그리고 해당 교집합을 반복하여, count에 추가하여 반환하면 됨.

# 천재의 답안인 1번은 
# 해당 최솟값에 sum함수를 추가하여 변수의 존재가 필요없게됨